asynction/types.py

```diff
@@ -393,6 +394,9 @@ def forge(type_: Type["Channel"], data: JSONMapping, forge: Forge) -> "Channel":
             x_handlers=forge(
                 type_.__annotations__["x_handlers"], data.get("x-handlers")
             ),
+            x_security=forge(
+                type_.__annotations__["x_security"], data.get("x-security", [])
+            ),
```

Comment:
The channel security requirements should be validated against the `components.securitySchemes`, similar to what we've done for the server requirements.